src/dmd/dsymbolsem.d

```diff
@@ -2841,6 +2902,11 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             sc.parent = ns;
             foreach (s; *ns.members)
             {
+                if (repopulateMembers)
```

Comment:
Is this necessary? The scopesym should not change by inserting mangleOnly namespaces and it seems dangerous to insert symbols into the symbol table again. Maybe calling s.setScope(sc) is enough.

Reply:
I didn't understand this code as well as I'd like. I can try your suggestion?
I'd love an expert to fix these couple of lines, they made me a touch nervous, but the unit tests all passed, so I had moderate confidence I worked it out right...